src/connections/api.js

```diff
@@ -0,0 +1,193 @@
+            this.#ws.send(JSON.stringify({ error: 'Failed to process command' }));
+          }
+
+        } catch (error) {
+          this.#logger.error(`Unexpected error handling message: ${error.message}`);
+        }
+      });
+
+      this.#ws.addEventListener('close', () => {
+        this.#logger.log('WebSocket connection to public API server closed');
+      });
+
+      this.#ws.addEventListener('error', (error) => {
+        this.#logger.error(`WebSocket error: ${error.message}`);
+      });
+    } catch (error) {
+      this.#logger.error(`Error creating WebSocket connection: ${error.message}`);
+    }
+  }
+
+  // Start the ping heartbeat
+  #startHeartbeat() {
+    this.#pingInterval = setInterval(() => {
+      if (this.#ws.readyState === WebSocket.OPEN) {
+        this.#ws.send(JSON.stringify({ type: 'ping' }));
+      }
+    }, 90000); // Send a ping every 90 seconds
+  }
+
+  // Send API command
+  async sendAPI(output) {
+    try {
+      const ws = this.controller ? this.controller.currentResponse : null;
+
+      if (!ws || ws.readyState !== WebSocket.OPEN) {
+        this.#logger.error('WebSocket connection not found or closed');
+        return;
+      }
+
+      try {
+        ws.send(JSON.stringify({ message: output }));
+        this.controller.currentResponse = null;
+      } catch (error) {
+        this.#logger.error(`Failed to send response: ${error.message}`);
+      }
+    } catch (error) {
+      this.#logger.error(`Unexpected error in sendAPI: ${error.message}`);
+    }
+  }
+
+  // Send broadcast message
+  async sendBroadcastMessage(message, type = 'frontend') {
+    try {
+      const ws = this.controller ? this.controller.ws : null;
+
+      if (!ws || ws.readyState !== WebSocket.OPEN) {
+        this.#logger.error('WebSocket connection not found or closed');
+        return;
+      }
+
+      try {
+        ws.send(JSON.stringify({ type: type, data: message }));
+      } catch (error) {
+        this.#logger.error(`Failed to send broadcast message: ${error.message}`);
+      }
+    } catch (error) {
+      this.#logger.error(`Unexpected error in sendBroadcastMessage: ${error.message}`);
+    }
+  }
+}
+
+// Global error handling
+process.on('unhandledRejection', (reason, promise) => {
+  console.error('Unhandled Rejection at:', promise, 'reason:', reason);
+});
+
+process.on('uncaughtException', (error) => {
+  console.error('Uncaught Exception:', error);
+});
+
+// Export a function that instantiates the API class with controller
+module.exports = (controller) => {
+  try {
+    const wsUrl = process.env.PUBLIC_WS_URL; // Get the Public WS server url
+    const secretKey = process.env.JWT_SECRET; // Get the JWT decrypt token
+    const wsKey = process.env.WS_SECRET_TOKEN; // Get the Websocket key token 
+    controller.connections.api = new API(wsUrl, wsKey, secretKey, controller); 
+  } catch (error) {
+    console.error('Failed to instantiate API:', error.message);
+  }
```

Comment:
Similarly here, not sure this should have a try/catch -- if it fails, it should fail

Reply:
I think my goal was to make sure nothing with the API takes down the entire chatbot as the API is not critical to the bots functioning. But if its better practice remove the try/catch we can do that.